sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -1378,19 +1399,26 @@ class RocksDBFileMapping {
    */
   private def getDfsFileForSave(
       fileManager: RocksDBFileManager,
-      localFileName: String,
+      localFile: File,
       versionToSave: Long): Option[RocksDBImmutableFile] = {
-    getDfsFileWithVersionCheck(fileManager, localFileName, _ >= versionToSave)
+    getDfsFileWithIncompatibilityCheck(
+      fileManager,
+      localFile.getName,
+      (dfsFileVersion, dfsFile) =>
+        // The DFS file is not the same as the file we want to save, either if
+        // the DFS file was added in the same or higher version, or the file size is different
+        dfsFileVersion >= versionToSave || dfsFile.sizeBytes != localFile.length()
```

Comment:
Just to make clear: would `dfsFile.sizeBytes` make another call to remote FS, or will it contain the info already?

Reply:
Already contains the info in dfsFile. See `RocksDBSstFile` case class.